src/client/app/components/kid-view/KidShop.jsx

```diff
@@ -17,28 +17,26 @@ class KidShop extends Component {
   }
 
   onSubmit() {
-    event.preventDefault();
-    var existingItems;
-		if (this.state.items == ''){
-			existingItems = [];
-		} else {
-			existingItems = this.state.items;
-		}
-    existingItems.push(this.state.item);
-    this.setState({items: existingItems});
+    if (this.state.item !== ''){
+      event.preventDefault();
```

Comment:
event.preventDefault typically used to prevent the default submission action and reload of an html form component.

Is this an actual form component. AKA does this do anything?
If YES, then why is it fired conditionally? You would want to prevent this default behavior from reloading the page your react-app is rendering on or otherwise if this is not needed remove it.